File: core/src/main/java/com/bing/lan/core/base/mapper/LogininfoMapper.java
package com.bing.lan.core.base.mapper;

import com.bing.lan.core.base.domain.Logininfo;

import org.apache.ibatis.annotations.Param;

import java.util.List;

public interface LogininfoMapper {

    int deleteByPrimaryKey(Long id);

    int insert(Logininfo record);

    Logininfo selectByPrimaryKey(Long id);

    List<Logininfo> selectAll();

    int updateByPrimaryKey(Logininfo record);

    int getCountByUsername(@Param("username") String username);

    Logininfo login(@Param("username") String username, @Param("password") String password);
}

File: core/src/main/kotlin/com/bing/lan/core/base/mapper/LogininfoMapper.kt
package com.bing.lan.core.base.mapper

import com.bing.lan.core.base.domain.Logininfo

import org.apache.ibatis.annotations.Param

interface LogininfoMapper {

    fun deleteByPrimaryKey(id: Long?): Int

    fun insert(record: Logininfo): Int

    fun selectByPrimaryKey(id: Long?): Logininfo

    fun selectAll(): List<Logininfo>

    fun updateByPrimaryKey(record: Logininfo): Int

    fun getCountByUsername(@Param("username") username: String, @Param("userType") userType: Int): Int

    fun login(@Param("username") username: String, @Param("password") password: String, @Param("userType") userType: Int): Logininfo
}